File: dkb/convert_dkb_amount.go
package dkb

import (
	"errors"
	"fmt"
	"strconv"
	"strings"
	"unicode"
)

func convertDkbAmount(input string) (string, error) {
	var builder strings.Builder

	for _, char := range input {
		if unicode.IsDigit(char) || char == rune("-"[0]) {
			builder.WriteRune(char)
		} else if char == rune(","[0]) {
			builder.WriteRune(rune("."[0]))
		}
	}

	value, err := strconv.ParseFloat(builder.String(), 64)
	if err != nil {
		return "", errors.New("Failed to parse amount " + input)
	}

	return fmt.Sprintf("%.2f", value), nil
}

File: dkb/convertDkbAmount.ts
export function convertDkbAmount(input: string): string {
  const cleanedInput = input.replace(/[^0-9,-]/g, "").replace(",", ".");
  const value = parseFloat(cleanedInput);

  if (isNaN(value)) {
    throw new Error(`Failed to parse amount ${input}`);
  }

  return value.toFixed(2);
}
